### PR TITLE
Added detailed lifecycle diagram and revised explanatory text details

### PR DESCRIPTION
**Pull request by ken-ebert**
This replaces PR #577 . I have added a diagram and simplified the text to reflect the lack of implied ordering of actions.

***
<a href="https://pr-preview.s3.amazonaws.com/ken-ebert/vc-data-model/pull/624.html" title="Last updated on May 15, 2019, 4:48 AM UTC (ab5ab13)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/624/d873334...ken-ebert:ab5ab13.html" title="Last updated on May 15, 2019, 4:48 AM UTC (ab5ab13)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1855,73 +1855,73 @@ <h2>Advanced Concepts</h2>
 this section explores more complex topics about <a>verifiable credentials</a>.
       </p>
 
-      <section class="informative">
```
Comment (msporny) [inline index.html]: Start of `section` tag was deleted, but closing tag wasn't, resulting in mismatched section tags throughout the document. Please fix mismatched tags.

Comment (msporny) [reply index.html]: On second thought, I don't understand why this section was deleted and merged into the "Advanced Concepts" section, as it has more to do w/ lifecycle than advanced concepts. My guess is that this was an accidental deletion? In that case, this PR should probably put the section header back in.

File: index.html [hunk 2]
```diff
@@ -1855,73 +1855,73 @@ <h2>Advanced Concepts</h2>
...
 
-        <ol>
-          <li>
-The <a>verifier</a> defines its policies for accepting
-<a>verifiable credentials</a> from <a>holders</a> for its supported actions.
-          </li>
+        <figure id="life-cycle-details">
+          <img style="margin: auto; display: block; width: 85%;"
+         src="diagrams/ecosystemdetail.svg" alt="diagram showing how
+         credentials flow from issuer to holder, from holder to holders, and
```
Comment (TallTed) [inline index.html]: `flow from issuer to holder, from holder to holders,`
-> `flow from issuer to holder, and optionally from one holder to another; and how`

File: index.html [hunk 3]
```diff
@@ -1855,73 +1855,73 @@ <h2>Advanced Concepts</h2>
...
 <a>verifiable presentation</a> and <a>verifiable credentials</a>.
+This should include checking the <a>credential status</a> for revocation
+of <a>verifiable credentials</a>. Verification occurs some time after a
+<a>verifiable presentation</a>.
           </li>
-          <li>
-The <a>verifier</a> <a>verifies</a> that the <a>holder</a> possesses the
-<a>verifiable credentials</a>.
+         <li>
+An <a>issuer</a> <a>revokes</a> a <a>verifiable credential</a>.
```
Comment (TallTed) [inline index.html]: `An <a>issuer</a> <a>revokes</a>`
-> `An <a>issuer</a> may <a>revoke</a>`

File: index.html [hunk 4]
```diff
@@ -1855,73 +1855,73 @@ <h2>Advanced Concepts</h2>
...
           </li>
           <li>
-The <a>verifier</a> decides whether to accept the <a>verifiable credentials</a>
-for the requested action, taking into account its policy, the <a>holder</a>,
-and the contents of the <a>verifiable credentials</a>.
+A <a>holder</a> deletes a <a>verifiable credential</a>.
```
Comment (TallTed) [inline index.html]: `A <a>holder</a> deletes`
-> `A <a>holder</a> may delete`

File: index.html [hunk 5]
```diff
@@ -1855,73 +1855,73 @@ <h2>Advanced Concepts</h2>
...
+<a>present</a>, and <a>verify</a>.
         <p>
 This specification does not define any protocol for transfering
 <a>verifiable credentials</a> or <a>verifiable presentations</a>, but assuming
 other specifications do specify how they are transferred between entities, then
-this Verifiable Credential Data Model is:
+this Verifiable Credential Data Model is directly applicable.
+        </p>
+        <p>
+This specification does not define an authorization framework or the
```
Comment (TallTed) [inline index.html]: `framework or the`
-> `framework nor the`

**Comment by TallTed** [comment]
Several minor tweaks to the text. I wish the SVG were visible in the preview. If it's as we drafted over the weekend, I'm probably fine with it ... but I must reserve final judgement.

File: index.html (hunk 1 above)
Comment (ken-ebert) [reply index.html]: Accidental deletion. Restoring the section header.

File: index.html (hunk 2 above)
Comment (ken-ebert) [reply index.html]: Good wording. I'll update it.

File: index.html (hunk 3 above)
Comment (ken-ebert) [reply index.html]: Agreed.

File: index.html (hunk 4 above)
Comment (ken-ebert) [reply index.html]: Agreed.

File: index.html (hunk 5 above)
Comment (ken-ebert) [reply index.html]: Agreed.

**Comment by ken-ebert** [comment]
@TallTed 
> Several minor tweaks to the text. I wish the SVG were visible in the preview. If it's as we drafted over the weekend, I'm probably fine with it ... but I must reserve final judgement.

I made some small adjustments in the same graphic file I shared with you previously. Again, I welcome your feedback. The link is below.

@msporny Anyone should be able to view the diagram at this link:
https://docs.google.com/drawings/d/1kOMAWPFVvN6LwwTJP-Mqx0xZtVY3ozoBk5MXYEnllCM/edit?usp=sharing

**Comment by David-Chadwick** [comment]
Nice diagram. My only slight worry is that this presents the picture that VCs are designed to be swapped between holders multiple times, back and forth, without restriction, when in practice this is not going to be the case. Most physical VCs today are issued to the subject and kept by the subject. And work in ISO on electronic IDs and driving licenses is ensuring that the subject does not pass his/her VC onto another holder.

**Comment by TallTed** [comment]
I think restriction of holder-to-holder handoffs is a protocol question. Such handoffs may and will happen, so the model must allow for it. I don't think anything in the current diagram indicates that such handoffs are *necessary*, only *possible*.

**Comment by David-Chadwick** [comment]
I have a number of changes to suggest to this text as follows.

1. Issuance must occur before any other actions involving this credential can occur.  >
Issuance always occurs before any other actions involving this credential. 

2. A holder presents one or more of its verifiable credentials to a verifier in a verifiable presentation. >
A holder presents one or more of its verifiable credentials to a verifier, optionally inside a verifiable presentation. 

3. This should include checking the credential status for revocation of verifiable credentials. >
This should include checking the credential status for revocation of the verifiable credentials. 

4. Verification occurs some time after a verifiable presentation.  > null (i.e. delete)

5. The simplest useful sequence of actions envisioned would be issue, present, and verify. >
The most common sequence of actions are envisioned to be: issuer issues to subject, subject presents to verifier, and verifier verifies.

**Comment by ken-ebert** [comment]
@David-Chadwick Do you have a suggested specific change to the diagram that I could make to communicate the exceptional rather than ordinary nature of transfers?

**Comment by David-Chadwick** [comment]
If I were to change the diagram I would delete the top set of holders and turn the double headed arrow that points to them into a 3/4 circle arrow that points to holders underneath the top holder in the centre of the diagram.

**Comment by ken-ebert** [comment]
Thank you @David-Chadwick for your suggested improvements to the text and diagram. I have revised the text and diagram to reflect your suggestions.

The updated diagram is located at the same link for review prior to the pull request being merged.: https://docs.google.com/drawings/d/1kOMAWPFVvN6LwwTJP-Mqx0xZtVY3ozoBk5MXYEnllCM/edit?userstoinvite=macted@gmail.com&ts=5cd5b33d&actionButton=1

File: index.html [hunk 6]
```diff
@@ -1861,67 +1861,73 @@ <h3>Lifecycle Details</h3>
...
 
-        <ol>
-          <li>
-The <a>verifier</a> defines its policies for accepting
-<a>verifiable credentials</a> from <a>holders</a> for its supported actions.
-          </li>
+        <figure id="life-cycle-details">
+          <img style="margin: auto; display: block; width: 85%;"
+         src="diagrams/ecosystemdetail.svg" alt="diagram showing how
+         credentials flow from issuer to holder, from and optionally
```
Comment (TallTed) [inline index.html]: `holder, from and`
-> `holder, and`

File: index.html [hunk 7]
```diff
@@ -1861,67 +1861,73 @@ <h3>Lifecycle Details</h3>
...
-The <a>verifier</a> defines its policies for accepting
-<a>verifiable credentials</a> from <a>holders</a> for its supported actions.
-          </li>
+        <figure id="life-cycle-details">
+          <img style="margin: auto; display: block; width: 85%;"
+         src="diagrams/ecosystemdetail.svg" alt="diagram showing how
+         credentials flow from issuer to holder, from and optionally 
+         from one holder to another; and how
+         presentations flow from holder to verifier, where
+         three parties can use information from a logical
```
Comment (TallTed) [inline index.html]: `three parties`
-> `all parties`

File: index.html [hunk 8]
```diff
@@ -1861,67 +1861,73 @@ <h3>Lifecycle Details</h3>
...
+          <figcaption style="text-align: center;">
+            The roles and information flows for this specification.
+          </figcaption>
+        </figure>
+
+        <ul>
           <li>
-The <a>issuer</a> issues a <a>verifiable credential</a> to the <a>holder</a>.
+An <a>issuer</a> <a>issues</a> a <a>verifiable credential</a> to a <a>holder</a>.
+Issuance always occurs before any other actions involving this credential.
```
Comment (TallTed) [inline index.html]: `involving this credential`
-> `involving a credential`

File: index.html [hunk 9]
```diff
@@ -1861,67 +1861,73 @@ <h3>Lifecycle Details</h3>
...
           <li>
-The final <a>holder</a> presents its <a>verifiable credentials</a> to the
-<a>verifier</a> in a <a>verifiable presentation</a>, requesting a supported
-action.
+A <a>holder</a> presents one or more of its <a>verifiable credentials</a> to a
+<a>verifier</a>, optionally inside a <a>verifiable presentation</a>.
           </li>
           <li>
-The <a>verifier</a> <a>verifies</a> the authenticity of the
+A <a>verifier</a> <a>verifies</a> the authenticity of the
```
Comment (TallTed) [inline index.html]:
```
the authenticity of the
<a>verifiable presentation</a> and <a>verifiable credentials</a>.
```
->
```
the authenticity of the presented
<a>verifiable presentations</a> and/or <a>verifiable credentials</a>.
```

File: index.html [hunk 10]
```diff
@@ -1861,67 +1861,73 @@ <h3>Lifecycle Details</h3>
...
-more than once. Such step-recurrence may be immediate or at any later
-point in the sequence as performed.
+The order of the actions above is not fixed, and some actions might be taken
+more than once. Such action-recurrence may be immediate or at any later
+point.
         </p>
-
+        
+        <p>
+The most comon sequence of actions are envisioned to be
```
Comment (TallTed) [inline index.html]: `sequence of actions are envisioned`
-> `sequence of actions is envisioned`

File: index.html [hunk 11]
```diff
@@ -1861,67 +1861,73 @@ <h3>Lifecycle Details</h3>
...
 This specification does not define any protocol for transfering
 <a>verifiable credentials</a> or <a>verifiable presentations</a>, but assuming
 other specifications do specify how they are transferred between entities, then
-this Verifiable Credential Data Model is:
+this Verifiable Credential Data Model is directly applicable.
+        </p>
+        <p>
+This specification does not define an authorization framework nor the
+decisions that a <a>verifier</a> might make after <a>verifying</a> a 
+<a>verifiable presentation</a>, taking into account the <a>holder</a>,
```
Comment (TallTed) [inline index.html]: `<a>verifiable presentation</a>, taking`
-> `<a>verifiable credential</a> or <a>verifiable presentation</a>, taking`

**Comment by TallTed** [comment]
I've made several further changes to the diagram, and suggested several minor text tweaks. I think we're close, if not there, with these.

File: index.html (hunk 6 above)
Comment (ken-ebert) [reply index.html]: done.

File: index.html (hunk 7 above)
Comment (ken-ebert) [reply index.html]: Done.

File: index.html (hunk 8 above)
Comment (ken-ebert) [reply index.html]: done.

File: index.html (hunk 9 above)
Comment (ken-ebert) [reply index.html]: done.

File: index.html (hunk 10 above)
Comment (ken-ebert) [reply index.html]: done. good catch.

File: index.html (hunk 11 above)
Comment (ken-ebert) [reply index.html]: done.

**Comment by ken-ebert** [comment]
Thanks, Ted for your detailed review and suggestions. I think the diagram changes add clarity to the status check and the number of times actions can occur.

**Pull request merged**